lib/src/models/store_model_snapshot.dart

```diff
@@ -0,0 +1,163 @@
+        images: storeModel.images,
+        townCode: storeModel.townCode,
+        latlong: latlong,
+      );
+
+  StoreModel toStoreModel({required ChainStoreModel chainStore}) => StoreModel(
+        owner: chainStore.name,
+        category: chainStore.category,
+        description: chainStore.description,
+        name: '${chainStore.name} - $name',
+        address: address,
+        phone: phone.ext.phoneFormatValue,
+        images: images,
+        townCode: townCode,
+        createdAt: createdAt,
+        updatedAt: updatedAt,
+        isApproved: true,
+        documentId: id ?? '',
+      );
+
+  final String name;
+  final String address;
+  final String phone;
+  final int townCode;
+  final String? id;
+  final List<String> images;
+  @JsonKey(includeFromJson: false, includeToJson: false)
+  @override
+  final String documentId;
+
+  @JsonKey(
+    toJson: FirebaseTimeParse.dateTimeToTimestamp,
+    fromJson: FirebaseTimeParse.datetimeFromTimestamp,
+    defaultValue: DateTime.now,
+  )
+  final DateTime? createdAt;
+  @JsonKey(
+    toJson: FirebaseTimeParse.dateTimeToTimestamp,
+    fromJson: FirebaseTimeParse.datetimeFromTimestamp,
+    defaultValue: DateTime.now,
+  )
+  final DateTime? updatedAt;
+
+  @JsonKey(
+    fromJson: GeoParser.fromJsonGeoPoint,
+    toJson: GeoParser.toJsonGeoPoint,
+    name: 'latlong',
+  )
+  final GeoPoint latlong;
+
+  @override
+  List<Object?> get props {
+    return [
+      name,
+      address,
+      phone,
+      id,
+      createdAt,
+      updatedAt,
+      latlong,
+    ];
+  }
+
+  @override
+  StoreModelSnapshot fromFirebase(DocumentSnapshot<Map<String, dynamic>> json) {
+    if (json.data() == null) return StoreModelSnapshot.empty();
+    return _$StoreModelSnapshotFromJson(json.data()!)
+        .copyWith(documentId: json.id);
+  }
+
+  @override
+  StoreModelSnapshot fromJson(Map<String, dynamic> json) =>
+      _$StoreModelSnapshotFromJson(json);
+
+  @override
+  Map<String, dynamic> toJson() => _$StoreModelSnapshotToJson(this);
+
+  factory StoreModelSnapshot.fromJson(Map<String, dynamic> json) =>
+      _$StoreModelSnapshotFromJson(json);
+
+  StoreModelSnapshot copyWith({
+    String? name,
+    String? address,
+    String? phone,
+    int? townCode,
+    String? id,
+    List<String>? images,
+    String? documentId,
+    DateTime? createdAt,
+    DateTime? updatedAt,
+    GeoPoint? latlong,
+  }) {
+    return StoreModelSnapshot(
+      name: name ?? this.name,
+      address: address ?? this.address,
+      phone: phone ?? this.phone,
+      townCode: townCode ?? this.townCode,
+      id: id ?? this.id,
+      images: images ?? this.images,
+      documentId: documentId ?? this.documentId,
+      createdAt: createdAt ?? this.createdAt,
+      updatedAt: updatedAt ?? this.updatedAt,
+      latlong: latlong ?? this.latlong,
+    );
+  }
+
+  @override
+  String toString() {
+    final name = 'Şube adı: ${this.name}';

```

Comment:
Açıklamanın ingilizce olması daha iyi olabilir

Reply:
Admin tarafında localize olmadığı için direkt toString i bu şekilde ezmiştim, ingilizce değil de toStringi modelden ezmek yerine kullanacağımız yerde manuel de basabiliriz